earthfile2llb/interpreter.go

```diff
@@ -1433,6 +1436,34 @@ func (i *Interpreter) handleArg(ctx context.Context, cmd spec.Command) error {
 	return nil
 }
 
+func (i *Interpreter) handleLet(ctx context.Context, cmd spec.Command) error {
+	if i.pushOnlyAllowed {
+		return i.pushOnlyErr(cmd.SourceLocation)
+	}
+	var opts letOpts
+	argsCpy := getArgsCopy(cmd)
+	args, err := parseArgs("LET", &opts, argsCpy)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse SET args")
+	}
+	if len(args) != 3 || args[1] != "=" {
+		return hint.Wrap(errInvalidSyntax, "LET requires a variable assignment, e.g. LET foo = bar")
+	}
+
+	key := args[0]
+	baseVal := args[2]
+	val, err := i.expandArgs(ctx, baseVal, true, false)
+	if err != nil {
+		return i.wrapError(err, cmd.SourceLocation, "failed to expand LET value %q", baseVal)
```

Comment:
nit: If not appropriate, feel free to ignore or click "resolved".

My apologies as it's my first time looking at the interpreter code - how would this show up in a failed run?
Perhaps we should include the key here as well?

```suggestion
		return i.wrapError(err, cmd.SourceLocation, "failed to expand LET value %q for key %q", baseVal, key)
```

Reply:
I was basically trying to match the error messages from `ARG` here. The nice thing about `i.wrapError` is that it will add the file name, line number, and column number that the error occurred on - so it will be pretty obvious which `LET` line failed to parse. Including the key might be nice, but feels largely unnecessary since the line number is included already (and I'd like to stay consistent with `ARG`).